packages/react-router/__tests__/data-memory-router-test.tsx

```diff
@@ -2392,7 +2392,7 @@ describe("<DataMemoryRouter>", () => {
       `);
 
       barValueDfd.resolve("LAZY");
-      await waitFor(() => !screen.getByText(/Loading.../));
+      await waitFor(() => screen.getByText(/oops is not defined/));
```

Comment:
I need to put together a little isolated test case here.  Somehow RTL would resolve the above `await` that `Loading...` was _not_ present, but then the actual HTML _would_ contain loading.  Adding a setTimeout fixes the issue, so it seems to be something related to concurrent mode maybe.  Simply waiting for the error to be displayed instead of the lack of a fallback also fixes the test.